tempesta_fw/ss_skb.c

```diff
@@ -601,19 +681,18 @@ __skb_fragment(struct sk_buff *skb, struct sk_buff *pskb, char *pspt,
  * so use kfree() to release memory.
  */
 int
-ss_skb_get_room(struct sk_buff *skb, char *pspt, unsigned int len, TfwStr *it)
+ss_skb_get_room(SsSkbList *head, struct sk_buff *skb,
+		char *pspt, unsigned int len, TfwStr *it)
 {
-	if (unlikely(len > PAGE_SIZE)) {
-		SS_WARN("Trying to get too large skb room, size=%u\n", len);
-		return -E2BIG;
-	}
+	int r = __skb_fragment(skb, NULL, pspt, len, it);
+	ss_skb_defraglist(head, skb);
```

Comment:
In the manipulations on the SKB it may be necessary to create a new SKB to hold extra fragments. The new SKB is created in `__extend_pgfrags()`. The new SKB has to be inserted into the SKB list for an HTTP message in correct place, after the current SKB that is being processed. That operation requires the list head. So the list head will have to be passed through the whole call chain in `ss_skb.c` all the way down to `__extend_pgfrags()`. 

Is that what we want to do?

Instead, I temporarily store it in SKB's `frag_list` as now we don't have data there. When the control is returned back to one of the top functions in the call chain that has the pointer to the list head, the SKB is moved to the SKB list for the HTTP message.